kedro/config/omegaconf_config.py

```diff
@@ -224,25 +226,19 @@ def __getitem__(self, key) -> dict[str, Any]:  # noqa: PLR0912
             else:
                 raise exc
 
-        # Destructively merge the two env dirs. The chosen env will override base.
-        common_keys = config.keys() & env_config.keys()
-        if common_keys:
-            sorted_keys = ", ".join(sorted(common_keys))
-            msg = (
-                "Config from path '%s' will override the following "
-                "existing top-level config keys: %s"
-            )
-            _config_logger.debug(msg, env_path, sorted_keys)
-
-        config.update(env_config)
+        merging_strategy = self.merge_strategy.get(key)
+        if merging_strategy == "soft":
+            resulting_config = self._soft_merge(config, env_config)
+        elif merging_strategy == "destructive" or not merging_strategy:
+            resulting_config = self._destructive_merge(config, env_config, env_path)
```

Comment:
```suggestion
        elif merging_strategy == "destructive" or not merging_strategy:
            resulting_config = self._destructive_merge(config, env_config, env_path)
        else:
             raise InvalidConfigurationError
```
Should we consider catching the invalid config? i.e. typo

Reply:
Or we can do a checking in the beginning -  `if merging_strategy in ("destructive", "soft")`

Reply:
Updated!